conans/client/rest/rest_client_v2.py

```diff
@@ -153,13 +145,10 @@ def _download_and_save_files(self, urls, dest_folder, files):
         for filename in sorted(files, reverse=True):
             resource_url = urls[filename]
             abs_path = os.path.join(dest_folder, filename)
+            os.makedirs(os.path.dirname(abs_path), exist_ok=True)  # filename in subfolder must exist
             downloader.download(url=resource_url, file_path=abs_path, auth=self.auth,
                                 verify_ssl=self.verify_ssl, retry=retry, retry_wait=retry_wait)
 
-    def _remove_recipe_files(self, ref, files):
```

Comment:
This is the origin of everything: Using revisions, no longer need to remove files in the recipes in the server, this never happens.